pvanalytics/quality/time.py

```diff
@@ -184,3 +184,63 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     # localize the shift_amount series to the timezone of the input
     shift_amount = shift_amount.tz_localize(event_times.index.tz)
     return shift_amount != 0, shift_amount
+
+
+def _has_dst(event_times, date, window, min_difference):
+    before = event_times[date - window:date - pd.Timedelta(days=1)]
+    after = event_times[date:date + window]
+    if len(before) == 0 or len(after) == 0:
+        raise ValueError(f"Insufficient data at {date}.")
+    before = before.dt.hour * 60 + before.dt.minute
+    after = after.dt.hour * 60 + after.dt.minute
+    return abs(before.mean() - after.mean()) > min_difference
+
+
+def has_dst(event_times, shift_dates, window=7, min_difference=45):
+    """Return true if `daynight_mask` appears to have daylight-savings shifts
+    at the dates in `shift_dates`.
+
+    Compares the mean event time in minutes since midnight over the
+    `window` days before and after each date in `shift_dates`. If the
+    difference is greater than `min_difference` then a shift has occurred
+    on that date.
+
+    Parameters
+    ----------
+    event_times : Series
+        Series with one timestamp for each day. The timestamp should
+        correspond to an event that occurs at roughly the same time on
+        each day, and shifts with daylight savings transitions. For example,
+        you may pass sunrise, sunset, or solar transit time.
+    shift_dates : list of datetime-like
+        Dates of expected daylight savings time shifts. Can be any type
+        that can be converted to a ``pandas.Timestamp`` by
+        :py:func:`pandas.to_datetime`.
+    window : int, default 7
+        Number of days before and after the shift date to consider. When
+        passing rounded timestamps in `event_times` it may be necessary to
+        use a smaller window. [days]
+    min_difference : int, default 45
+        Minimum difference between the mean event time before the shift
+        date and the mean event time after the event time. If the difference
+        is greater than `min_difference` a shift has occurred on that date.
+        [minutes]
+
+    Returns
+    -------
+    list of bool
+        Boolean indicating whether a DST shift was found at each date in
```

Comment:
should this Bool series be indexed by `event_times` rather than `shift_dates`? That's more a question about user expectation

Reply:
Now returns series indexed by daylight savings transition dates. I'm not 100% sold in that being the right behavior, but I think it is the best option. To determine if the series has dst at all, users can use `time.has_dst(x).any()`.